lhotse/recipes/mgb2.py

```diff
@@ -0,0 +1,238 @@
+    When all the manifests are available in the ``output_dir``, it will simply read and return them.
+
+    :param corpus_dir: Pathlike, the path of the data dir.
+    :param output_dir: Pathlike, the path where to write the manifests.
+    :param text_cleaning: Bool, if True, basic text cleaning is performed (similar to ESPNet recipe).
+    :param buck_walter: Bool, use BuckWalter transliteration
+    :param num_jobs: int, the number of jobs to use for parallel processing.
+    :param mer_thresh: int, filter out segments based on mer (Match Error Rate)
+    :return: a Dict whose key is the dataset part, and the value is Dicts with the keys 'audio' and 'supervisions'.
+
+    .. note::
+        Unlike other recipes, output_dir is not Optional here because we write the manifests
+        to the output directory while processing to avoid OOM issues, since it is a large dataset.
+
+    .. caution::
+        The `text_cleaning` option removes all punctuation and diacritics.
+    """
+    if not is_module_available("bs4"):
+        raise ValueError(
+            "To prepare MGB2 data, please 'pip install beautifulsoup4' first."
+        )
+
+    from bs4 import BeautifulSoup
+
+    corpus_dir = Path(corpus_dir)
+    assert corpus_dir.is_dir(), f"No such directory: {corpus_dir}"
+
+    dataset_parts = ["dev", "train", "test"]
+    manifests = {}
+
+    if output_dir is not None:
+        output_dir = Path(output_dir)
+        output_dir.mkdir(parents=True, exist_ok=True)
+        # Maybe the manifests already exist: we can read them and save a bit of preparation time.
+        manifests = read_manifests_if_cached(
+            dataset_parts=dataset_parts,
+            output_dir=output_dir,
+            prefix="mgb2",
+            suffix="jsonl.gz",
+            lazy=True,
+        )
+
+    for part in dataset_parts:
+        info(f"Processing MGB2 subset: {part}")
+        if manifests_exist(
+            part=part, output_dir=output_dir, prefix="mgb2", suffix="jsonl.gz"
+        ):
+            info(f"MGB2 subset: {part} already prepared - skipping.")
+            continue
+
+        # Read the recordings and write them into manifest. We additionally store the
+        # duration of the recordings in a dict which will be used later to create the
+        # supervisions.
+
+        output_dir = Path(output_dir)
+        corpus_dir = Path(corpus_dir)
+        if part == "test" or part == "dev":
+            (output_dir / part).mkdir(parents=True, exist_ok=True)
+            copy(
+                corpus_dir / part / "text.non_overlap_speech",
+                output_dir / part / "text",
+            )
+            copy(
+                corpus_dir / part / "segments.non_overlap_speech",
+                output_dir / part / "segments",
+            )
+            system(
+                f"cat {(corpus_dir / part / 'wav.scp')}\
+                 | sed 's:wav/:{(corpus_dir / part / 'wav')}/:g' > {path.join(output_dir, part,'wav.scp')}"
+            )
```

Comment:
This `system` call can also be changed to use python as:
```python
with open(corpus_dir / part / "wav.scp", "r") as f_in, open(output_dir / part / "wav.scp", "w") as f_out:
    for line in f_in:
        f_out.write(line.replace("wav/", f"{corpus_dir}/{part}/wav/")})
        f_out.write("\n")
```

Reply:
Done